### PR TITLE
[Mobile Payments] With TTP visible, onboarding error hides bottom rows on the payments hub

### LINKED ISSUE #8766: [Mobile Payments] With TTP visible onboarding error hides bottom rows on the payments hub
Image

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #8766

### Description

The PR:
* ~~Removes "learn more" from the payments hub. In order to make it work right it has to be moved inside of the Recycler View, and I personally don't think that it brings much to the table. The payments hub contains many things so it is not really clear what a user should "learn more" about. Context is missing here @AnirudhBhat  @samiuelson wdyt about that?~~ **Added Learn More as an item of the Recycler View**
* Adds padding so onboarding error doesn't hide last rows
* Fixes determination of "the same item" and "the same content" so there is no more blinking of the "manage card reader" row

### Testing instructions

* Open payments hub with onboarding completed. Notice that the "manage card reader" row didn't blink and there is small padding at the bottom
* Open payments hub with onboarding not completed. Notice that the "manage card reader" row didn't blink and there is  padding at the bottom enough to show all the rows behind the error pop up

#### Before

https://user-images.githubusercontent.com/4923871/230606786-79f6c017-e0b2-402a-a3e8-488893ef53e1.mp4

#### After

https://user-images.githubusercontent.com/4923871/230606791-ed505de1-ab2c-47b1-8697-5f806d1dfd95.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModel.kt [hunk 1]
```diff
@@ -485,69 +463,6 @@ class CardReaderHubViewModel @Inject constructor(
         data class ShowToast(@StringRes val message: Int) : CardReaderHubEvents()
     }
 
-    data class CardReaderHubViewState(
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/hub/CardReaderHubViewModel.kt:488]: Just moved to a separate file

**Comment by AnirudhBhat** [comment]
>Removes "learn more" from the payments hub. In order to make it work right it has to be moved inside of the Recycler View, and I personally don't think that it brings much to the table. The payments hub contains many things so it is not really clear what a user should "learn more" about. Context is missing here

Thanks for the ping @kidinov 

[Looking at the usage in tracks](https://mc.a8c.com/tracks/trends/?compareby=prop&eventname=woocommerceandroid_in_person_payments_learn_more_tapped&interval=day&prop=source&startdate=20230312&stat=uniques), it seems like quite a few people click on the learn more link in the payments hub screen.

Since the payments hub is the main entry point for all things IPP in our app, it kinda makes sense to have a learn more link there? The rest of the learn more links are buried deep under the IPP transaction screens and it would get noticed only by the merchants who are already aware of IPP.

I agree the payment hub screen seems to have quite a bit of things going but it all boils down to In-Person Payment eventually. Right? Just different ways of doing IPP (card reader vs Tap To Pay).

For someone with very little knowledge about IPP in WooCommerce app, payments hub seems to be a good place to learn more about IPP. WDYT?

In the future, we could have a separate section called `Information` along with the `Actions` and `Card Readers` sections. The `Information` section contains everything the merchant needs to understand more about IPP, TTP, fees, taxes related to IPP ...etc.

In short, I don't see any harm in leaving the learn more link in the payments hub as it is now since it seems like people are making use of it quite often. WDYT?

**Comment by kidinov** [comment]
@AnirudhBhat 

Thank you for the reply! 

> [Looking at the usage in tracks](https://mc.a8c.com/tracks/trends/?compareby=prop&eventname=woocommerceandroid_in_person_payments_learn_more_tapped&interval=day&prop=source&startdate=20230312&stat=uniques), it seems like quite a few people click on the learn more link in the payments hub screen.

That's only ~1% of the people who open this screen,  but I'll add it back for now and see how the screen evolves

**Review by samiuelson — APPROVED** [review]
LGTM! Thanks for adding these improvements! :shipit:
